### PR TITLE
Deveoper Guide for Orca

### PR DESCRIPTION
**Pull request by charlieJ107**
Add a Dockerfile of developer container and its document. With this Dockerfile, developer can create a out of box develop environment within single command.

### CODE REVIEW

**Comment by yangw1234** [comment]
any changes to https://github.com/intel-analytics/BigDL/blob/branch-2.0/python/orca/developer_guide.md?

**Comment by yangw1234** [comment]
also would you mind pushing the image to your personal repo so that others can try it?

**Comment by charlieJ107** [comment]
> any changes to https://github.com/intel-analytics/BigDL/blob/branch-2.0/python/orca/developer_guide.md? Like the prepare_env.sh path?

The `prepare_env.sh` has been added in https://github.com/intel-analytics/BigDL/commit/2823dc3ac8de8b260bd295c59dce6778e771fa65 . So the `deveoper_guid.md` seems no need to change any more.

**Comment by charlieJ107** [comment]
> also would you mind pushing the image to your personal repo so that others can try it?

Pre-build image has been pushed to Docker Hub since https://github.com/intel-analytics/BigDL/pull/3818/commits/3c1855f66166bb67aaee50a8556f346edc7ce39a.

File: python/orca/developer_container.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
+# BigDL orca Developer Container
+
+This docker image contain the basic development environment for BigDL-Orca. You can build a new container with the Dockerfile or simply start from a pre-build image. 
+
```
Comment (yangw1234) [inline python/orca/developer_container.md:4]: maybe first add a section on how to use this image? 
e.g.
```bash
docker pull vankyle2go/orca-dev:latest
docker run --name orca-dev -it vankyle2go/orca-dev:latest
```

**Comment by yangw1234** [comment]
LGTM

**Pull request merged**